### PR TITLE
[Concurrency] Drop the cloned code for Embedded around 'any Actor'

### PR DESCRIPTION
**Pull request by kubamracek**
Now that `any Actor` is usable in Embedded Swift, we can drop all this ifdef'd code.

### CODE REVIEW

**Comment by kubamracek** [comment]
@swift-ci please test

**Review by rauhul — APPROVED** [review]
Nice, love negative diffs

File: stdlib/public/Concurrency/DiscardingTaskGroup.swift [hunk 1]
```diff
@@ -184,159 +184,6 @@ public struct DiscardingTaskGroup {
     let _: Void? = try await _taskGroupWaitAll(group: _group, bodyError: nil)
   }
 
-// Clone the task-creation routines in Embedded Swift so that we don't
-// introduce an implicit use of `any Actor`.
-#if !$Embedded
```
Comment (benrimmington) [inline stdlib/public/Concurrency/DiscardingTaskGroup.swift:189]: Are you sure that the `#if !$Embedded` branch is the one to remove? It contains the `@isolated(any)` additions (introduced by commit a86b76a3e9ebe352e8d250927c211445ac692c26).

**Comment by kubamracek** [comment]
@swift-ci please test

**Review by ktoso — COMMENTED** [review]
Very happy to see some reduction of duplication for a change :) Thank you @kubamracek !

**Review by ktoso — APPROVED** [review]

**Pull request merged**